utils/files.go

```diff
@@ -0,0 +1,67 @@
+package utils
+
+import (
+	"bufio"
+	"log"
+	"os"
+	"strconv"
+)
+
+/*
+Read a file line-by-line and parse each line to an int
+
+TODO: is there a better composable "stream" interface for reading / parsing?
+
+cribbed from https://stackoverflow.com/questions/8757389/reading-a-file-line-by-line-in-go
+*/
+func ReadLines(path string) (<-chan string, error) {
+	file, err := os.Open(path)
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO: does returning a channel make sense, or is it better to pass one in?
+	results := make(chan string)
+	go func() {
+		defer file.Close()
+		defer close(results)
+
+		scanner := bufio.NewScanner(file)
+		for scanner.Scan() {
+			results <- scanner.Text()
+		}
+
+		if err := scanner.Err(); err != nil {
+			log.Fatal(err)
+		}
+	}()
+
+	return results, nil
+}
+
+/*
+Read a file line-by-line and parses each line to an int
+*/
+func ParseInts(path string) (<-chan int, error) {
+	lines, err := ReadLines(path)
+	if err != nil {
+		return nil, err
+	}
+	results := make(chan int)
```

Comment:
I'm using this "wrap a channel" approach a lot, I think I'm leaning on that where I would use `map` or `yield`. 

Does golang want me to not worry about streaming and just pass slices around instead?

Reply:
Seeing a lot of commentary on stack overflow, but not much in official docs. Seems like "iterator" or "generator" isn't really a common pattern.

My approach here is pretty close to stuff on http://www.golangpatterns.info/, but I'm not sure if I should trust that website

Reply:
replaced these generators with just slices, it seems a little simpler at the expense of theoretical memory pressure